pkg/console/operator/operator.go

```diff
@@ -251,3 +260,36 @@ func (c *consoleOperator) removeConsole(cr *operatorsv1.Console) error {
 
 	return utilerrors.FilterOut(utilerrors.NewAggregate(errs), errors.IsNotFound)
 }
+
+func (c *consoleOperator) handleConfigMapSync(operatorConfig *operatorsv1.Console) error {
+	c.CheckCustomLogoImageStatus(operatorConfig)
+	logoName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+	// add syncing for the custom logo config map
+	if logoName != "" {
+		return c.resourceSyncer.SyncConfigMap(
+			resourcesynccontroller.ResourceLocation{Namespace: api.OpenShiftConsoleNamespace, Name: logoName},
+			resourcesynccontroller.ResourceLocation{Namespace: api.OpenShiftConfigNamespace, Name: logoName},
+		)
+	}
+	return nil
+
+}
+
+func (c *consoleOperator) CheckCustomLogoImageStatus(operatorConfig *operatorsv1.Console) {
+	logo := operatorConfig.Spec.Customization.CustomLogoFile
+	if logo.Name != "" && logo.Key != "" {
+		//Check that configmap for custom logo exists
+		logoConfigMap, err := c.configMapClient.ConfigMaps(api.OpenShiftConfigNamespace).Get(logo.Name, metav1.GetOptions{})
+		if err != nil {
+			klog.Errorf("customLogo configmap not valid, %v", err)
```

Comment:
Do you ever see these `klog` calls?

Reply:
Meaning, should this func return an error that bubbles up? Or should the errors be exclusively in logs.

Reply:
Yes, I can see these statements in our log.

Reply:
I would prefer these stay in the log and the condition shows on the status of the operator config

Reply:
Look around though, I believe every function in the project that makes an API call returns an error which bubbles up.

Reply:
`Name` is the configmap name, which we should sync into `openshift-console` as `custom-logo`.
`Key` is the file name key.
The `<value>` of the key is the actual image.  There are a number of possible issues that could arise here, but I think we can ignore them for the most part. It could be `null`, or the `base64` encoded string may not be an image at all or a different image.  Lets not chase that rabbit unless we find reason to do so.